src/main/java/net/mcreator/element/types/Structure.java

```diff
@@ -25,14 +25,13 @@
 import net.mcreator.element.parts.procedure.Procedure;
 import net.mcreator.element.types.interfaces.ICommonType;
 import net.mcreator.workspace.elements.ModElement;
-import net.mcreator.workspace.references.ResourceReference;
 
 import java.util.Collection;
 import java.util.List;
 
 @SuppressWarnings("unused") public class Structure extends GeneratableElement implements ICommonType {
 
-	@ResourceReference("structure") public String structure;
+	public String structure;

```

Comment:
How will structure reference be obtained now?

Reply:
> At this point, **on hold** until #3456 and #4274.

Reply:
I think I can still leave comments, right?

Reply:
This is done to avoid merge conflicts as I'd need to fully check the refactored classes anyway.

Reply:
I see. You can still work on other parts. This PR digs deeply into the code so there will be merge conflicts all the time until we get this PR right and merge it, unfortunatelly